### PR TITLE
Add a command to push to OCI registries

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/roadmap/issues/391

## Checklist

- [x] Make yourself familiar with following readme sections:
    - [x] [Coding Standards](https://github.com/giantswarm/architect-orb#coding-guidelines).
    - [x] [Development](https://github.com/giantswarm/architect-orb#development).
    - [x] [Design and Goals](https://github.com/giantswarm/architect-orb#design-and-goals).
- [x] :warning: Update changelog in [CHANGELOG.md](https://github.com/giantswarm/architect-orb/tree/master/CHANGELOG.md).

### CODE REVIEW

File: src/commands/package-and-push-oci.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+parameters:
+  app_catalog:
```
Comment (MarcelMue) [inline src/commands/package-and-push-oci.yaml:2]: Should we set defaults on these?

Comment (kubasobon) [reply src/commands/package-and-push-oci.yaml:2]: I like this idea a lot. Done.

File: src/commands/package-and-push-oci.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      command: |
+          readonly app_catalog_name="$(cat .app_catalog_name)"
+          readonly reference="$(cat .reference)"
+
+          ret=1
+          tries=4
+          for i in $(seq 1 $tries) ; do
+            echo "====> Attempt $i: Running: helm push ../build/*.tgz $app_catalog_name"
+            set +e
+            helm push ../build/*.tgz $app_catalog_name
```
Comment (MarcelMue) [inline src/commands/package-and-push-oci.yaml:83]: Maybe a stupid question: Where does helm actually come from here? Is it just installed by default? And should we verify that the version is recent enough?

**Review by MarcelMue — APPROVED** [review]
Generally looking good to me - just a small question :)

File: src/commands/package-and-push-oci.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        mkdir build && helm package ./helm/<< parameters.chart >> --destination ./build
+  - when:
+      condition: << parameters.persist_chart_archive >>
+      steps:
+        - persist_to_workspace:
+            root: build
+            paths:
+              - "<< parameters.chart >>*.tgz"
+  - run:
+      name: Push chart archive to OCI registy app catalog
```
Comment (rossf7) [inline src/commands/package-and-push-oci.yaml:73]:
```suggestion
      name: Push chart archive to OCI registry app catalog
```

**Review by rossf7 — APPROVED** [review]
LGTM

File: src/commands/package-and-push-oci.yaml (hunk 2 above)
Comment (kubasobon) [reply src/commands/package-and-push-oci.yaml:83]: `architect-orb` uses `architect`'s Dockerfile as a base image. We install a bunch of binaries in that Dockerfile, including helm. I've recently updated `architect` (https://github.com/giantswarm/architect/pull/728/files), and `architect-orb` is already using the latest release.

**Comment by kubasobon** [comment]
> Thats good! TBH a solution where the git based pushing would be moved into a separate command and then making the pushing command/mechanism configurable in the existing jobs would be less confusing, but this is also fine for me.

Bundling the two sounds like a solid idea. I'll add it to the proposed PR as well.

**Review by MarcelMue — APPROVED** [review]
It still makes sense to me - I struggle a bit to understand how this looks like in a project in the end, but that's a minor issue.

**Comment by kubasobon** [comment]
@MarcelMue This PR adds a new command set for pushing charts to OCI registries (two new commands). It also adds two switches to the `push-to-app-catalog` job. `push_to_appcatalog`, enabled by default, runs existing commands and publishes charts to GitHub. `push_to_oci_registry`, disabled by default, runs new commands and publishes charts to the AzureCR OCI registry. 

If we merge this and craft a release, everything will remain the same due to the defaults. If we flip the switches we can change behavior to push to OCI as well or push to it exclusively.

Do we want `push_to_oci_registry` right away?

**Comment by MarcelMue** [comment]
I think it makes sense to release first with the switch disabled and then try it in select repositories. WDYT?

**Comment by kubasobon** [comment]
I agree :man_scientist:

**Comment by kubasobon** [comment]
It works!
Screenshot_20220421_120317
Screenshot_20220421_120259

**Comment by kubasobon** [comment]
@ubergesundheit PTAL. I think this covers all your requests. The only thing I had to do to enable this is add the `push_to_oci_registry: true` to my test branch.

```yaml
workflows:
  version: 2
  build-and-push:
    jobs:
      - architect/push-to-app-catalog:
          context: "architect"
          executor: "app-build-suite"
          name: "package and push chart to app catalog"
          app_catalog: "giantswarm-catalog"
          app_catalog_test: "giantswarm-test-catalog"
          chart: "hello-world-app"
          # Trigger job on git tag.
          filters:
            tags:
              only: /^v.*/
          push_to_oci_registry: true
```

**Comment by ubergesundheit** [comment]
Thanks :) Some addition to the [docs](https://github.com/giantswarm/architect-orb/blob/master/docs/job/push-to-app-catalog.md) would have been nice

**Comment by kubasobon** [comment]
Sure, on it.

**Pull request merged**